src/openms/source/SYSTEM/File.cpp

```diff
@@ -450,11 +450,21 @@ namespace OpenMS
   String File::getTempDirectory()
   {
     Param p = getSystemParameters();
-    if (p.exists("temp_dir") && String(p.getValue("temp_dir")).trim() != "")
+    String dir;
+    if (getenv("OPENMS_TMPDIR") != 0)
+    {
+      dir = getenv("OPENMS_TMPDIR");
+    }
+    else if (p.exists("temp_dir") && String(p.getValue("temp_dir")).trim() != "")
+    {
+      dir = p.getValue("temp_dir");
+    }
+    else
     {
-      return p.getValue("temp_dir");
+      dir = String(QDir::tempPath());
     }
-    return String(QDir::tempPath());
+    dir.ensureLastChar('/');
```

Comment:
potentially dangerous on Windows if the path already ends in '\'

Reply:
unfortunately, there are other places in File.cpp where this is used as well

Reply:
@cbielow are you sure that is the case here, it seems oddly strange that all functions here use `/` as path separetors. I think it works because we use Qt which expects `/` as path separator and translates this to the appropriate path on the platform. See for example this docu: http://doc.qt.io/qt-4.8/qfileinfo.html and also https://stackoverflow.com/questions/37821181/qprocess-running-process-whose-path-contain-spaces

so I think its actually required to use `/` -- however we should document that in our `File.h`

Reply:
the reason its dangerous is because we do not have control over the value of `dir`. The user could specify it as `c:\temp\` in his environment and we would add `/` giving `c:\temp\/`, which could be problematic. Its simply because `.ensureLastChar()` is a function which is completely unaware of paths and platform separators. The code before was fine, because there was no user input and using `/` on windows in principle does no harm.
Or did I miss something?